android/src/main/java/com/oblador/keychain/cipherStorage/CipherStorageKeystoreRSAECB.java

```diff
@@ -0,0 +1,339 @@
+            }
+        } catch (KeyStoreException e) {
+            throw new KeyStoreAccessException("Failed to access Keystore", e);
+        } catch (Exception e) {
+            throw new KeyStoreAccessException("Unknown error " + e.getMessage(), e);
+        }
+    }
+
+    private byte[] encryptString(Key key, String service, String value) throws CryptoFailedException {
+        try {
+            Cipher cipher = Cipher.getInstance(ENCRYPTION_TRANSFORMATION);
+            cipher.init(Cipher.ENCRYPT_MODE, key);
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+            // encrypt the value using a CipherOutputStream
+            CipherOutputStream cipherOutputStream = new CipherOutputStream(outputStream, cipher);
+            cipherOutputStream.write(value.getBytes("UTF-8"));
+            cipherOutputStream.close();
+            return outputStream.toByteArray();
+        } catch (Exception e) {
+            throw new CryptoFailedException("Could not encrypt value for service " + service, e);
+        }
+    }
+
+    private String decryptBytes(Key key, byte[] bytes) throws CryptoFailedException, UserNotAuthenticatedException {
+        try {
+            Cipher cipher = Cipher.getInstance(ENCRYPTION_TRANSFORMATION);
+            ByteArrayInputStream inputStream = new ByteArrayInputStream(bytes);
+            // read the initialization vector from the beginning of the stream
+            cipher.init(Cipher.DECRYPT_MODE, key);
+            // decrypt the bytes using a CipherInputStream
+            CipherInputStream cipherInputStream = new CipherInputStream(
+                    inputStream, cipher);
+            ByteArrayOutputStream output = new ByteArrayOutputStream();
+            byte[] buffer = new byte[1024];
+            while (true) {
+                int n = cipherInputStream.read(buffer, 0, buffer.length);
+                if (n <= 0) {
+                    break;
+                }
+                output.write(buffer, 0, n);
+            }
+            return new String(output.toByteArray(), Charset.forName("UTF-8"));
+        } catch (UserNotAuthenticatedException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new CryptoFailedException("Could not decrypt bytes", e);
+        }
+    }
+
+    private KeyStore getKeyStoreAndLoad() throws KeyStoreException, KeyStoreAccessException {
+        try {
+            KeyStore keyStore = KeyStore.getInstance(KEYSTORE_TYPE);
+            keyStore.load(null);
+            return keyStore;
+        } catch (NoSuchAlgorithmException | CertificateException | IOException e) {
+            throw new KeyStoreAccessException("Could not access Keystore", e);
+        }
+    }
+
+    @NonNull
+    private String getDefaultServiceIfEmpty(@NonNull String service) {
+        return service.isEmpty() ? DEFAULT_SERVICE : service;
+    }
+
+    @Override
+    public boolean getRequiresCurentActivity() {
```

Comment:
I made a typo w/ this method name: `getRequiresCurentActivity` -> `getRequiresCurrentActivity`